src/xcode/ENA/ENA/Source/Services/Exposure Submission/ExposureSubmissionService.swift

```diff
@@ -152,21 +152,19 @@ class ENAExposureSubmissionService: ExposureSubmissionService {
 					with: self.symptomsOnset
 				)
 
-				let unencryptedCheckinsEnabled = self.appConfigurationProvider.featureProvider.value(for: .unencryptedCheckinsEnabled)
+				let unencryptedCheckinsEnabled = AppFeatureProvider(appConfig: appConfig).value(for: .unencryptedCheckinsEnabled)
```

Comment:
👍

Reply:
This change would not take into account the settings from the dev-menu. Please see the decorater pattern implemented to add the dev-menu functionality. Maybe we should change the pattern. Because its not obvious.

Reply:
Ah, thanks for that hint! I've now reversed that change and instead made sure that `self.appConfigurationProvider.featureProvider` is always up-to-date whenever the app config publisher comes back. Maybe it makes sense to get rid of the `AppFeatureProvider(appConfig: appConfig)` again as the publisher is now more reliable?